lib/stack_master/cli.rb

```diff
@@ -206,10 +206,11 @@ def execute!
             region = args[0]
           end
 
-          execute_if_allowed_account(allowed_accounts) do
+          success = execute_if_allowed_account(allowed_accounts) do
             StackMaster.cloud_formation_driver.set_region(region)
-            StackMaster::Commands::Delete.perform(region, stack_name)
+            StackMaster::Commands::Delete.perform(region, stack_name).success?
           end
+          @kernel.exit false unless success
```

Comment:
`stack_master delete` is updated to exit with status `1` if the command object is not successful, or a disallowed AWS account is used.